app/src/state/editor.ts

```diff
@@ -108,29 +143,227 @@ export const saveDocument = createAsyncThunk<void, void, { state: RootState }>(
+      state.selection = null;
+    },
+    goLeft: (state) => {
+      assertSome(state);
+      const item = getCurrentItem(state.document.content, state.currentTime, true);
+      assertSome(item);
+      state.currentTime = item.absoluteStart;
+      state.selection = null;
+    },
+    goRight: (state) => {
+      assertSome(state);
+      const iter = skipToTime(state.currentTime, documentIterator(state.document.content), true);
+      iter.next();
+      const item = iter.next().value;
+      assertSome(item);
+      state.currentTime = item.absoluteStart;
+      state.selection = null;
+    },
+
+    unselect: (state) => {
+      assertSome(state);
+      state.selection = null;
+      state.selectionStartItem = null;
+      state.mouseSelection = false;
+    },
+    selectLeft: (state) => {
+      assertSome(state);
+      const selectionInfo = getSelectionInfo(state.selection, state.selectionStartItem);
+      if (!selectionInfo || !state.selection) {
+        const item = getCurrentItem(state.document.content, state.currentTime, true);
+        assertSome(item);
+        state.selection = { start: item.absoluteStart, length: item.end - item.start };
+        state.selectionStartItem = item;
+      } else {
+        const { leftEnd, rightEnd, currentEndLeft } = selectionInfo;
+        if (currentEndLeft) {
+          const item = getCurrentItem(state.document.content, leftEnd, true);
+          assertSome(item);
+          state.selection.length = rightEnd - item.absoluteStart;
+          state.selection.start = item.absoluteStart;
+        } else {
+          const item = getCurrentItem(state.document.content, rightEnd, true);
+          assertSome(item);
+          state.selection.length = item.absoluteStart - leftEnd;
+        }
+      }
+    },
+    selectRight: (state) => {
+      assertSome(state);
+      const selectionInfo = getSelectionInfo(state.selection, state.selectionStartItem);
+      if (!selectionInfo || !state.selection) {
+        const item = getCurrentItem(state.document.content, state.currentTime, false);
+        assertSome(item);
+        state.selection = { start: item.absoluteStart, length: item.end - item.start };
+        state.selectionStartItem = item;
+      } else {
+        const { leftEnd, rightEnd, currentEndRight } = selectionInfo;
+        if (currentEndRight) {
+          const item = getCurrentItem(state.document.content, rightEnd);
+          assertSome(item);
+          const itemEnd = item.absoluteStart + (item.end - item.start);
+          state.selection.length = itemEnd - leftEnd;
+        } else {
+          const item = getCurrentItem(state.document.content, leftEnd);
+          assertSome(item);
+          const itemEnd = item.absoluteStart + (item.end - item.start);
+          state.selection.length = rightEnd - itemEnd;
+          state.selection.start = itemEnd;
+        }
+      }
+    },
+    mouseSelectionStart: (state, arg: PayloadAction<TimedParagraphItem>) => {
```

Comment:
I assume the bug is somewhere here: If I select a word using the mouse, but only select part of it, the word is shown as fully selected until the mouse-up, at which point the selection is lost. IMHO the selection should stay after the mouseup

https://user-images.githubusercontent.com/6545097/133343589-dbc1dfe1-448a-431a-85d9-cb1769b1f353.mov

Reply:
This still doesn't work. From what I can tell the issue is that `mouseSelectionEnd` and `setTime` are both dispatched from the `click`-eventhandler. On my machine `setTime` is then processed *after* `mouseSelectionEnd`. So first `mouseSelectionEnd` sets the `state.mouseSelection` to false, which `setTime` checks and because of that it removes the selection